game/state/city/vehiclemission.cpp

```diff
@@ -1402,6 +1402,23 @@ void VehicleMission::update(GameState &state, Vehicle &v, unsigned int ticks, bo
 		}
 		case MissionType::GotoLocation:
 		case MissionType::Land:
+			if (!finished || v.currentBuilding == nullptr)
```

Comment:
What happens if you select multiple vehicles/agents on foot to investigate a sighting? IIRC the OG xcom waited for everyone to arrive

Reply:
Just noticed you mentioned this exact case in the comment above :)

I'd prefer if we fix that up before merging though, unless it ends up so hilariously complicated I'd try not to leave known holes.

Reply:
Wouldn't a semaphore-like approach work?
You can set `investigate` to `n` when sending `n` vehicles, then decrease the counter as they arrive to the building. When decreasing from 1 to 0 you can push the commence investigation event :)